graphql_compiler/compiler/ir_lowering_sql/__init__.py

```diff
@@ -0,0 +1,264 @@
+                                interface in the key is the most-derived common supertype
+                                of every GraphQL type in the "value" GraphQL union.
+                                Recursive expansion of type equivalence hints is not performed,
+                                and only type-level correctness of this argument is enforced.
+                                See README.md for more details on everything this parameter does.
+                                *****
+                                Be very careful with this option, as bad input here will
+                                lead to incorrect output queries being generated.
+                                *****
+
+    Returns:
+        SqlTree object containing SqlNodes organized in a tree structure.
+    """
+    query_path_to_location_info = {
+        location.query_path: location_info
+        for location, location_info in query_metadata_table.registered_locations
+    }
+    # Perform lowering passes over IR blocks
+    construct_result = ir_blocks.pop()
+    construct_result = lower_construct_result(construct_result)
+    block_index_to_location = get_block_index_to_location_map(ir_blocks)
+    ir_blocks = lower_unary_transformations(ir_blocks)
+    ir_blocks = lower_context_field_existence(ir_blocks)
+    ir_blocks = lower_optional_fields(
+        ir_blocks, block_index_to_location, query_path_to_location_info)
+    query_path_to_node = {}
+    query_path_to_filter = defaultdict(list)
+    query_path_to_tag_fields = defaultdict(list)
+    tree_root = None
+    for index, block in enumerate(ir_blocks):
+        if isinstance(block, (blocks.Recurse, blocks.Traverse, blocks.QueryRoot)):
+            location = block_index_to_location[index]
+            location_info = query_metadata_table.get_location_info(location)
+            query_path = location.query_path
+            if tree_root is None:
+                if not isinstance(block, blocks.QueryRoot):
+                    raise AssertionError(u'Encountered QueryRoot but tree root is already set.')
+                tree_root = SqlNode(block=block, query_path=query_path)
+                query_path_to_node[query_path] = tree_root
+            else:
+                parent_location = location_info.parent_location
+                parent_query_path = parent_location.query_path
+                parent_node = query_path_to_node[parent_query_path]
+                child_node = SqlNode(block=block, query_path=query_path)
+                parent_node.add_child_node(child_node)
+                query_path_to_node[query_path] = child_node
+        elif isinstance(block, blocks.Filter):
+            for context_field in get_tag_fields(block.predicate):
+                query_path_to_tag_fields[context_field.location.query_path].append(context_field)
+            query_path_to_filter[query_path].append((block, query_path))
+        else:
```

Comment:
I'm worried that this is a bug waiting to happen. At some point, we may add a new block, for which it's not clear that `continue` is the correct decision to make here.

I'd much prefer an explicit `elif ...: continue` (bonus points for a comment explaining why continuing is the right thing there) followed by `else: raise AssertionError(...)`